File: Printpress.UI/Printpress_Angular/src/app/features/order/models/orderSellingItem/order-selling-item-upsert.dto.ts
import { IObjectState } from '../../../../core/models/i-object-state';

export interface OrderSellingItemUpsertDto extends IObjectState {
    id: string;
    name: string;
    inventoryItemId?: string;
    isInventoryItem: boolean;
    quantity: number;
    price: number;
}

File: Printpress.Application/Orders/DTO/Order/Upsert/OrderSellingItemUpsertDTO.cs
using Printpress.Domain;
using System.Text.Json.Serialization;

namespace Printpress.Application;

public class OrderSellingItemUpsertDTO
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public Guid? InventoryItemId { get; set; }
    public bool IsInventoryItem { get; set; }
    public int Quantity { get; set; }
    public decimal Price { get; set; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public TrackingState ObjectState { get; set; }
}
